R/method-differentialAbundance.R

```diff
@@ -116,85 +64,182 @@ setMethod("differentialAbundance", signature("AbsoluteAbundanceData", "Comparato
-        dds <- DESeq2::DESeqDataSetFromMatrix(countData = counts,
-                                              colData = sampleMetadata,
-                                              design = as.formula(paste0("~",comparatorColName)),
-                                              tidy = FALSE)
-
-        # Estimate size factors before running deseq to avoid errors about 0 counts
-        geoMeans = apply(DESeq2::counts(dds), 1, function(x){exp(sum(log(x[x > 0]), na.rm=T) / length(x))})
-        dds <- DESeq2::estimateSizeFactors(dds, geoMeans = geoMeans)
-
-        # Run DESeq
-        deseq_output <- DESeq2::DESeq(dds)
-      })
-
-      if (veupathUtils::is.error(deseq_output)) {
-        veupathUtils::logWithTime(paste0('Differential abundance FAILED with parameters recordIdColumn=', recordIdColumn, ', method =', method, ', verbose =', verbose), verbose)
-        stop()
-      }
-
-      # Extract deseq results
-      deseq_results <- DESeq2::results(deseq_output)
-
-      # Format results for easier access
-      statistics <- data.frame(log2foldChange = deseq_results$log2FoldChange,
-                               pValue = deseq_results$pvalue,
-                               adjustedPValue = deseq_results$padj,
-                               pointID = rownames(counts))
-
-
-    } else if (identical(method, 'ANCOMBC')) {
-
-      se <- TreeSummarizedExperiment::TreeSummarizedExperiment(list(counts = counts), colData = sampleMetadata)
-
-      # Currently getting this error: Error in is.infinite(o1) : default method not implemented for type 'list'
-      # Ignoring for now.
-      output_abs = ANCOMBC::ancombc2(data = se, assay_name = "counts", tax_level = NULL,
-                  fix_formula = comparatorColName, rand_formula = NULL,
-                  p_adj_method = "holm", prv_cut=0,
-                  group = comparatorColName)
-
+      statistics <- deseq(data, comparator)
+#    } else if (identical(method, 'ANCOMBC')) {
+#
+#      se <- TreeSummarizedExperiment::TreeSummarizedExperiment(list(counts = counts), colData = sampleMetadata)
+#
+#      # Currently getting this error: Error in is.infinite(o1) : default method not implemented for type 'list'
+#      # Ignoring for now.
+#      output_abs = ANCOMBC::ancombc2(data = se, assay_name = "counts", tax_level = NULL,
+#                  fix_formula = comparatorColName, rand_formula = NULL,
+#                  p_adj_method = "holm", prv_cut=0,
+#                  group = comparatorColName)
+#
+    } else if (identical(method, 'MaAsLin2')) {
+      statistics <- maaslin(data, comparator)
     } else {
-      stop('Unaccepted differential abundance method. Accepted methods are "DESeq" and "ANCOMBC".')
+      stop('Unaccepted differential abundance method. Accepted methods are "DESeq" and "Maaslin".')
     }
     veupathUtils::logWithTime(paste0('Completed method=',method,'. Formatting results.'), verbose)
     
+    # TODO make sure i understood/ did this right..
```

Comment:
yup it's the dropped taxa. But i didn't want to name it "taxa" because it could be pathways, genes, etc.

Reply:
just want to check - is this clear now?